### PR TITLE
Json change record & serialization to TEvApplyReplicationChanges::TChange KIKIMR-20673

### PR DESCRIPTION
**Pull request by CyberROFL**
### Changelog entry 

...

### Changelog category 

* Not for changelog (changelog entry is not required)

### Additional information

...

### CODE REVIEW

File: ydb/core/tx/replication/service/json_change_record.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+            auto res = NFormats::MakeCell(cells.emplace_back(), value, it->second.Type, pool, error);
+            Y_ABORT_UNLESS(res);
+        }
+
+        upsert.SetData(TSerializedCellVec::Serialize(cells));
+    } else if (JsonBody.Has("erase")) {
+        record.MutableErase();
+    } else {
+        Y_ABORT("Malformed json record");
+    }
```
Comment (Enjection) [inline ydb/core/tx/replication/service/json_change_record.cpp:77]: Maybe it is better to let caller decide what to do in this case?

Comment (Enjection) [reply ydb/core/tx/replication/service/json_change_record.cpp:77]: In general it is okay, as far as we doing the same in other parts of code.

**Review by Enjection — CHANGES_REQUESTED** [review]

File: ydb/core/tx/datashard/change_record.h [hunk 2]
```diff
@@ -54,7 +62,28 @@ class TChangeRecord: public NChangeExchange::TChangeRecordBase {
 
 class TChangeRecordBuilder: public NChangeExchange::TChangeRecordBuilder<TChangeRecord, TChangeRecordBuilder> {
 public:
-    using NChangeExchange::TChangeRecordBuilder<TChangeRecord, TChangeRecordBuilder>::TChangeRecordBuilder;
+    using TBase::TBase;
+
+    explicit TChangeRecordBuilder(EKind kind)
+        : TBase()
```
[the remark this answers is not in the document]
Comment (CyberROFL) [reply ydb/core/tx/datashard/change_record.h:68]: yes

File: ydb/core/tx/replication/service/json_change_record.cpp (hunk 1 above)
Comment (CyberROFL) [reply ydb/core/tx/replication/service/json_change_record.cpp:77]: for now, record MUST be serializable.

**Review by Enjection — APPROVED** [review]

**Pull request merged**